app/src/test/java/org/fossasia/openevent/app/presenter/LoginPresenterTest.java

```diff
@@ -136,4 +143,37 @@ public void shouldNotAccessView() {
         Mockito.verifyNoMoreInteractions(loginView);
     }
 
+    @Test
+    public void shouldAttachEmailOnLoginSuccessfully() {
+        String authToken = "testToken";
+
+        Mockito.when(loginModel.login(email, password))
+            .thenReturn(Observable.just(new LoginResponse(authToken)));
+
+        Mockito.when(utilModel.getStringSet(Constants.SHARED_PREFS_SAVED_EMAIL, null))
+            .thenReturn(savedEmails);
+
+        loginPresenter.attach(loginView);
+        loginPresenter.login(email, password);
+
+        Mockito.verify(loginView).attachEmails(savedEmails);
+    }
+
+    @Test
+    public void shouldAttachEmailAutomatically() {
+        Mockito.when(utilModel.getStringSet(Constants.SHARED_PREFS_SAVED_EMAIL, null)).thenReturn(savedEmails);
```

Comment:
If the user is logged in, why to verify that the adapter of the autocompleteview is set in attachemails method?